### PR TITLE
Some misc cleanups for matrix

### PR DESCRIPTION
**Pull request by sylee957**
#### Brief description of what is fixed or changed

Some trivial changes made in matrix.

I've made every non square check with using `is_square` property
and has_floats check with `has()` property 

There already exist methods using exactly identical logics

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by sylee957** [comment]
```
2019-01-29T02:16:37.3786826Z ________________________________________________________________________________
2019-01-29T02:16:37.3787011Z _____________ sympy/matrices/tests/test_commonmatrix.py:test_trace _____________
2019-01-29T02:16:37.3787160Z Traceback (most recent call last):
2019-01-29T02:16:37.3787674Z   File "/opt/hostedtoolcache/Python/2.7.15/x64/lib/python2.7/site-packages/sympy-1.4.dev0-py2.7.egg/sympy/matrices/tests/test_commonmatrix.py", line 547, in test_trace
2019-01-29T02:16:37.3787875Z     assert M.trace() == 14
2019-01-29T02:16:37.3788511Z   File "/opt/hostedtoolcache/Python/2.7.15/x64/lib/python2.7/site-packages/sympy-1.4.dev0-py2.7.egg/sympy/matrices/common.py", line 1829, in trace
2019-01-29T02:16:37.3789921Z     if not self.is_square:
2019-01-29T02:16:37.3790232Z AttributeError: 'OperationsOnlyMatrix' object has no attribute 'is_square'
2019-01-29T02:16:37.3790324Z ________________________________________________________________________________
2019-01-29T02:16:37.3790383Z _____________ sympy/matrices/tests/test_commonmatrix.py:test_power _____________
2019-01-29T02:16:37.3790428Z Traceback (most recent call last):
2019-01-29T02:16:37.3790772Z   File "/opt/hostedtoolcache/Python/2.7.15/x64/lib/python2.7/site-packages/sympy-1.4.dev0-py2.7.egg/sympy/matrices/tests/test_commonmatrix.py", line 704, in test_power
2019-01-29T02:16:37.3790831Z     assert (A**5)[:] == (6140, 8097, 10796, 14237)
2019-01-29T02:16:37.3791141Z   File "/opt/hostedtoolcache/Python/2.7.15/x64/lib/python2.7/site-packages/sympy-1.4.dev0-py2.7.egg/sympy/core/decorators.py", line 132, in binary_op_wrapper
2019-01-29T02:16:37.3791223Z     return func(self, other)
2019-01-29T02:16:37.3791525Z   File "/opt/hostedtoolcache/Python/2.7.15/x64/lib/python2.7/site-packages/sympy-1.4.dev0-py2.7.egg/sympy/matrices/common.py", line 2052, in __pow__
2019-01-29T02:16:37.3791579Z     if not self.is_square:
2019-01-29T02:16:37.3791977Z AttributeError: 'ArithmeticOnlyMatrix' object has no attribute 'is_square'
```

Some primitive matrix classes don't have property `is_square`, while `rows` and `cols` are defined in every matrices

**Comment by sylee957** [comment]
I've reverted the ones from `common.py`, unfortunately the tests are too strict for that.

File: sympy/matrices/common.py [hunk 1]
```diff
@@ -1826,7 +1826,7 @@ def trace(self):
         5
 
         """
-        if not self.is_square:
+        if not self.rows == self.cols:
```
Comment (jksuom) [inline sympy/matrices/common.py]: Or `self.rows != self.cols`?

**Comment by oscargus** [comment]
Would it make sense to add a property `is_nonempty` (and/or `is_empty`). I reacted on the `self.row < 1` which probably means `is_empty` where the latter reads better imho. Not sure how it affects the primitive classes that were mentioned (didn't follow the code progress).

**Comment by sylee957** [comment]
@oscargus 
Though it would be a good idea to add new properties for the purpose,
However, I find that it's a bit complicated to work with matrix properties, because I think that they were not already made consistently.

For example, you can see some are property and some are not being a property 
```
from sympy import *
a = Matrix.eye(100)
a.is_symmetric() # function
a.is_hermitian # property
```
And I think that it there is no possible way to make a property work both as a function and property, at least I find.
I think it may be opened up as a discussion

**Comment by sylee957** [comment]
I have opened up a issue #15883, if you have some thoughts about it

**Pull request merged**